### PR TITLE
WIP: Add LFS lock and unlock and show locks list

### PR DESCRIPTION
**Pull request by raptoravis**
Fixes #

## Proposed changes

- 
- 
- 

## Test methodology 

- 
- 
- 

## Test environment(s) 

- GIT 
- Windows 

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/Settings/AppSettings.cs [hunk 1]
```diff
@@ -1537,6 +1537,12 @@ public static bool UseFormCommitMessage
             set => SetBool("UseFormCommitMessage", value);
         }
 
+        public static bool UseFormLocksMessage
+        {
+            get => GetBool("UseFormLocskMessage", true);
```
Comment (pmiossec) [inline GitCommands/Settings/AppSettings.cs:1542]: Typo here

File: GitCommands/Git/GitModule.cs [hunk 2]
```diff
@@ -4153,6 +4153,43 @@ public string GetDescribe(ObjectId commitId)
             return output;
         }
 
+        public void LfsLock(IReadOnlyList<string> files)
+        {
+            if (files.Count == 0)
```
Comment (pmiossec) [inline GitCommands/Git/GitModule.cs:4158]: The code of this method is the same than the Unlock. Could you factories to a common method taking the git parameters as an input?

File: GitUI/FilterRevisionsHelper.cs [hunk 3]
```diff
@@ -86,6 +88,8 @@ public FilterRevisionsHelper(ToolStripTextBox textBox, ToolStripDropDownButton d
 
             _NO_TRANSLATE_showFirstParentButton.Checked = AppSettings.ShowFirstParent;
 
+            // _NO_TRANSLATE_showLocks.Checked = true;
```
Comment (pmiossec) [inline GitUI/FilterRevisionsHelper.cs:91]: remove comment?

File: GitUI/FilterRevisionsHelper.cs [hunk 4]
```diff
@@ -97,6 +101,9 @@ public FilterRevisionsHelper(ToolStripTextBox textBox, ToolStripDropDownButton d
             };
             _NO_TRANSLATE_showFirstParentButton.Click += delegate { _NO_TRANSLATE_revisionGrid.ShowFirstParent(); };
             _NO_TRANSLATE_revisionGrid.ShowFirstParentsToggled += delegate { _NO_TRANSLATE_showFirstParentButton.Checked = AppSettings.ShowFirstParent; };
+            _NO_TRANSLATE_showLocks.Click += delegate { _NO_TRANSLATE_revisionGrid.ShowLocks(); };
+
+            // _NO_TRANSLATE_revisionGrid.ShowLocks += delegate { _NO_TRANSLATE_showLocks.Checked = true; };
```
Comment (pmiossec) [inline GitUI/FilterRevisionsHelper.cs:106]: remove also?

File: GitCommands/Git/GitCommandHelpers.cs [hunk 5]
```diff
@@ -735,6 +735,99 @@ private static string RemoveWarnings(string statusString)
...
+            var lockFiles = new List<GitItemStatus>();
+
+            var files = lockedFilesString.Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            for (int n = 0; n < files.Length; n++)
+            {
+                string line = files[n];
+
+                var items = ParseLockedInfo(line);
+
+                char x = '?';
```
Comment (pmiossec) [inline GitCommands/Git/GitCommandHelpers.cs:749]: Move `x` outside of the loop as a `const` with a meaningful name, please.

File: GitCommands/Git/GitCommandHelpers.cs [hunk 6]
```diff
@@ -735,6 +735,99 @@ private static string RemoveWarnings(string statusString)
             return trimmedStatus;
         }
 
+        public static IReadOnlyList<GitItemStatus> GetLockedFileList(string lockedFilesString)
+        {
+            var lockFiles = new List<GitItemStatus>();
+
+            var files = lockedFilesString.Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            for (int n = 0; n < files.Length; n++)
```
Comment (pmiossec) [inline GitCommands/Git/GitCommandHelpers.cs:743]: Convert to a foreach.

File: GitCommands/Git/GitModule.cs [hunk 7]
```diff
@@ -4153,6 +4153,43 @@ public string GetDescribe(ObjectId commitId)
...
+            // See https://referencesource.microsoft.com/#system/services/monitoring/system/diagnosticts/Process.cs,1952
+            _gitExecutable.RunBatchCommand(new GitArgumentBuilder("lfs unlock", false)
+                .BuildBatchArgumentsForFiles(files));
+        }
+
+        public IReadOnlyList<GitItemStatus> LfsLockedFiles()
+        {
+            var output = _gitExecutable.GetOutput(new GitArgumentBuilder("lfs locks", false));
+
+            var result = GitCommandHelpers.GetLockedFileList(output).ToList();
```
Comment (pmiossec) [inline GitCommands/Git/GitModule.cs:4188]: remove use of temporary variable `result`.

**Comment by pmiossec** [comment]
Oh and please commit translations. See for details: https://github.com/gitextensions/gitextensions/wiki/Translations#release-translations

**Comment by RussKie** [comment]
Please add a comprehensive commit message explaining the reasoning behind
the change, how it works, pros and cons of the implementations,
screenshots, and explain how it was tested.

**Comment by pmiossec** [comment]
@raptoravis I have added 'WIP' in the subject as it seems it's a work in progress. Feel free to remove it when you think it's finished and ready to merge.

Also, please rebase instead of merging if you need to sync with `master`.
I also don't get why you update the assemblyInfo version. There is no need of that...

And please, fix the translation problem as soon as possible so that the CI result provide good feedback.

It would also be good that you provide some feedback to the team because we see commit pushed and that makes little sense for us....

**Comment by pmiossec** [comment]
@raptoravis Why did you closed this PR?
It's an interesting feature...

But as a big feature, perhaps you need to discuss a little with the team on how to integrate it in the UI.

**Pull request closed without merging**